src/mlpack/methods/reinforcement_learning/training_config.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file training_config.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the implementation of TrainingConfig class,
+ * which contains hyper-parameters for training RL agent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+#define MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+
+namespace mlpack {
+namespace rl {
+
+class TrainingConfig
+{
+ public:
+  TrainingConfig() : stepLimit(0), gradientLimit(40), doubleQLearning(false)
+  { /* Nothing to do here. */ }
+
+  /**
+   * Get and modify the number workers.
+   * This is valid only for async RL agent.
+   **/
+  size_t NumOfWorkers() const { return numOfWorkers; }
+  size_t& NumOfWorkers() { return numOfWorkers; }
```

Comment:
I'd call this `NumWorkers()` just to be consistent with other mlpack code that looks like `NumX()` and not `NumOfX()`.